From 7b530e623dd6a10f88eff797768b247ba405bac9 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 16 Oct 2013 13:07:26 +0800
Subject: [PATCH 34/50] ENGR00283397: cracked frame if take pictures during
 1080p recording

Each time switch to recording, startStream will called,
then call releaseAllRecordingFramesLocked,
This will Force-freed buffer status.
So a buffer maybe updated while it's under encoding.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 services/camera/libcameraservice/api1/client2/StreamingProcessor.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/services/camera/libcameraservice/api1/client2/StreamingProcessor.cpp b/services/camera/libcameraservice/api1/client2/StreamingProcessor.cpp
index 6076dae..231dc17 100644
--- a/services/camera/libcameraservice/api1/client2/StreamingProcessor.cpp
+++ b/services/camera/libcameraservice/api1/client2/StreamingProcessor.cpp
@@ -428,12 +428,17 @@ status_t StreamingProcessor::startStream(StreamType type,
 
     Mutex::Autolock m(mMutex);
 
+
+    //Force-freed buffer will make status chaos.
+    //So a buffer maybe updated while it's under encoding.
+#if 0
     // If a recording stream is being started up, free up any
     // outstanding buffers left from the previous recording session.
     // There should never be any, so if there are, warn about it.
     if (isStreamActive(outputStreams, mRecordingStreamId)) {
         releaseAllRecordingFramesLocked();
     }
+#endif
 
     ALOGV("%s: Camera %d: %s started, recording heap has %d free of %d",
             __FUNCTION__, mId, (type == PREVIEW) ? "preview" : "recording",
-- 
1.8.0

